pkg/runner/step_docker.go

```diff
@@ -58,7 +58,10 @@ func (sd *stepDocker) runUsesContainer() common.Executor {
 		if err != nil {
 			return err
 		}
-		entrypoint := strings.Fields(eval.Interpolate(step.With["entrypoint"]))
+		entrypoint, err := shellquote.Split(eval.Interpolate(step.With["entrypoint"]))
+		if err != nil {
+			return err
+		}
```

Comment:
Can we get rid of any split of entrypoint? ( My suggestion might make the linter unhappy )

```suggestion
		entrypoint := []string{eval.Interpolate(step.With["entrypoint"])}
```
My actions/runner local tool supports this:
```
[uses-docker-url / test] Running: Run ubuntu.fritz.box/entrypointwithspaces
|##[group]Run docker://ubuntu.fritz.box/entrypointwithspaces
|with:
|  entrypoint: /path with spaces/node
|  args: -v
|##[endgroup]
|##[command]C:\Program Files\Docker\Docker\resources\bin\docker.EXE inspect --format="{{.Os}}/{{.Architecture}}" ubuntu.fritz.box/entrypointwithspaces
|linux/amd64
|##[command]C:\Program Files\Docker\Docker\resources\bin\docker.EXE run --name ubuntufritzboxentrypointwithspaces_d4ca9e --label 8f0fd6 --workdir /github/workspace --rm -e INPUT_ENTRYPOINT -e INPUT_ARGS -e HOME -e GITHUB_JOB -e GITHUB_SERVER_URL -e GITHUB_API_URL -e GITHUB_GRAPHQL_URL -e GITHUB_WORKFLOW -e GITHUB_REPOSITORY -e GITHUB_SHA -e GITHUB_REPOSITORY_OWNER -e GITHUB_REF -e GITHUB_REF_PROTECTED -e GITHUB_REF_TYPE -e GITHUB_REF_NAME -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_ACTOR -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_RETENTION_DAYS -e GITHUB_RUN_ATTEMPT -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e GITHUB_ACTION_REPOSITORY -e GITHUB_ACTION_REF -e GITHUB_PATH -e GITHUB_ENV -e GITHUB_STEP_SUMMARY -e RUNNER_OS -e RUNNER_ARCH -e RUNNER_NAME -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e ACTIONS_CACHE_URL -e GITHUB_ACTIONS=true -e CI=true --entrypoint "/path with spaces/node" -v "/var/run/docker.soc...
|v16.13.1
[uses-docker-url / test] Succeeded: Run ubuntu.fritz.box/entrypointwithspaces
```

Reply:
Will that work with nested quotations, escaping, mixed quote style, etc.?

Reply:
In the docs it says that `entrypoint:` only accepts a single string: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepswithentrypoint - so I suppose the suggestion from @ChristopherHX looks correct

Reply:
it's annoying and confusing because actions are allowed to use more than one argv for `entrypoint` but it's not allowed in workflows